### PR TITLE
* Updated doc for launching the cluster using improved EC2 scripts.

### PR DESCRIPTION
**Pull request by ashetkar**
## Changes proposed in this pull request
Updated doc for launching the cluster using improved EC2 scripts.

## Patch testing
Tested manually

## ReleaseNotes.txt changes
NA

## Other PRs 
https://github.com/SnappyDataInc/snappy-cloud-tools/pull/12

### CODE REVIEW

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 1]
```diff
@@ -208,31 +205,35 @@ Here:
 
 * `<your-keyfile-path>` refers to the path to the key file.
 
-* `<action>` refers to the action to be performed. You must first launch your cluster using the `launch` action. The `start` and `stop` actions can be used to manage the nodes in the cluster.
+* `<action>` refers to the action to be performed. Some of the available actions are `launch`, `destroy`, `stop`, `start` and `reboot-cluster`.
+Use `launch` action to create a new cluster while `stop` and `start` actions work on existing clusters.
 
 By default, the script starts one instance of a locator, lead, and server each.
-The script identifies each cluster by its unique cluster name (you provided) and internally ties members (locators, leads, and stores/servers) of the cluster with EC2 security groups.
+The script identifies each cluster by its unique cluster name (that you provide) and internally ties members (locators, leads, and stores/servers) of the cluster with EC2 security groups - whose names are also derived from the cluster name.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: The script identifies each cluster by its unique cluster name that you provide and internally ties the members (locators, leads, and stores/servers) of the cluster with EC2 security groups, whose names are derived from the cluster name.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 2]
```diff
@@ -208,31 +205,35 @@ Here:
...
 
-The  names and details of the members are automatically derived from the provided cluster name. When running the script you can also specify properties like the number of stores and region.
+When running the script you can also specify options like the number of stores in the cluster and region where the EC2 instances should be launched.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: When running the script, you can also specify options such as the number of stores in the cluster and the region where the EC2 instances should be launched.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 3]
```diff
@@ -208,31 +205,35 @@ Here:
...
 **Example**
 
 ```pre
 ./snappy-ec2 -k my-ec2-key -i ~/my-ec2-key.pem --stores=2 --with-zeppelin --region=us-west-1 launch my-cluster
 ```
 
-In the above example, you are launching a SnappyData cluster named **my-cluster** with 2 stores (or servers). The locator is available in security group named **my-cluster-locator** and the store/server are available in **my-cluster-store**.
+The above example will launch a SnappyData cluster named **my-cluster** with 2 stores (or servers). The locator will be associated with security group named **my-cluster-locator** and the servers will be associated with **my-cluster-store** security group.
+
+The cluster will be launched in the N. California (us-west-1) region on AWS and it'll also have an Apache Zeppelin server running on the instance where the lead is running.</br>
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: The cluster is launched in the **N. California (us-west-1)** region on AWS and has an Apache Zeppelin server running on the instance where the lead is running.</br>

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 4]
```diff
@@ -208,31 +205,35 @@ Here:
...
 
-The cluster is launched in the N. California (us-west-1) region on AWS and starts an Apache Zeppelin server on the instance where the lead is running.</br>
 The example assumes that you have the key file (my-ec2-key.pem) in your home directory for EC2 Key Pair named 'my-ec2-key'.
 
 !!! Note
-	By default, the cluster is launched in the US East (N. Virginia) region on AWS. To launch the cluster in a specific region ensure that you set the region property `--region=`.
+	By default, the cluster is launched in the US East (N. Virginia) region on AWS. To launch the cluster in a specific region use option `--region`.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: By default, the cluster is launched in the **US East (N. Virginia)** region on AWS. To launch the cluster in a specific region use option `--region`.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 5]
```diff
@@ -291,12 +299,13 @@ When you start a cluster, it uses the existing EC2 instances associated with the
 ./snappy-ec2 -k my-ec2-key -i ~/my-ec2-key.pem start cluster-name
 ```
 !!! Note
-	The start command (or launch command with `--resume` option) ignores the `--locators`, `--leads`, or `--stores` options, and launches the SnappyData cluster on existing instances. The configuration files however, are read if they are present in the location mentioned above. You therefore must ensure that every time you use the `start` command, the number of entries in configuration files are equal to the number of instances in their respective security group.
+	The `start` command (or `launch` command with `--resume` option) ignores the `--locators`, `--leads`, or `--stores` options, and launches the SnappyData cluster on existing instances.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: The `start` command, or `launch` command with `--resume` option, ignores the `--locators`, `--leads`, or `--stores` options and launches the SnappyData cluster on existing instances.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 6]
```diff
@@ -291,12 +299,13 @@ When you start a cluster, it uses the existing EC2 instances associated with the
...
+	The configuration options, however, are read if they are provided, overriding those provided when the cluster was launched/started previously.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: However, the configuration options are read, if they are provided, by overriding those that are provided when the cluster was launched or started previously.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 7]
```diff
@@ -291,12 +299,13 @@ When you start a cluster, it uses the existing EC2 instances associated with the
...
 
 <a id="ec2addservertocluster"></a>
 #### Adding Servers to the Cluster
 
-This is not yet fully supported using the script. You may have to manually launch an instance with `(cluster-name)-stores` group, and then use launch command with the `--resume` option.
+This is not yet supported using the script. You may have to manually launch an instance with `(cluster-name)-stores` group, and then use `launch` command with the `--resume` option.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: This is not yet supported using the script. You must manually launch an instance with `(cluster-name)-stores` group and then use `launch` command with the `--resume` option.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 8]
```diff
@@ -322,20 +331,21 @@ Destroying a cluster permanently destroys all the data on the local instance sto
...
 ./snappy-ec2 -k my-ec2-key -i ~/my-ec2-key.pem destroy cluster-name
 ```
-This retains the security groups created for this cluster. To delete this, use it with `--delete-` group option.
+This also deletes the security groups created for this cluster.
 
 <a id="ec2startclustrezeppelin"></a>
 #### Starting Cluster with Apache Zeppelin
 
-Optionally, you can start an instance of Apache Zeppelin server with the cluster. [Apache Zeppelin](https://zeppelin.apache.org/) is a web-based notebook that enables interactive notebook. The Zeppelin server is launched on the same EC2 instance where the lead node is running.
+Optionally, you can start an instance of Apache Zeppelin server with the cluster. [Apache Zeppelin](https://zeppelin.apache.org/) provides a web-based interactive notebook that is pre-configured for you to talk to the SnappyData cluster. The Zeppelin server is launched on the same EC2 instance where the lead node is running.
```
Comment (lizygeogy) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: Suggestion: 
[Apache Zeppelin](https://zeppelin.apache.org/) provides a web-based interactive notebook that is pre-configured to communicate with SnappyData cluster.

File: docs/install/setting_up_cluster_on_amazon_web_services.md [hunk 9]
```diff
@@ -242,42 +247,50 @@ This sections covers the following:
...
-The important difference here is that, instead of the host names of the locator, lead or store, you have to write {{LOCATOR_N}}, {{LEAD_N}} or {{SERVER_N}} in these files, respectively. N stands for Nth locator, lead or server. The script replaces these with the actual host name of the members when they are launched.
+<a id="ec2custombuild"></a>
+#### Using custom build
 
-The sample configuration files for a cluster with 2 locators, 1 lead and 2 stores are given below:
+This script by default uses the SnappyData build available on the GitHub releases page to launch the cluster.
+To select a version of the build available on GitHub, use option `--snappydata-version`.
 
-*locators*
+You can also provide your own SnappyData build to the script to launch the cluster.
```
Comment (jramnara) [inline docs/install/setting_up_cluster_on_amazon_web_services.md]: I would specifically tell the reader that this is the option you should use if you want to use the enterprise version of the product. Download enterprise product from 'link'...

**Comment by jramnara** [comment]
On the docs ... given the new option where the user can input the build ...

I would specifically tell the reader that this is the option you should use if you want to use the enterprise version of the product. Download enterprise product from 'link'...

File: docs/install/setting_up_cluster_on_amazon_web_services.md (hunk 9 above)
Comment (ashetkar) [reply docs/install/setting_up_cluster_on_amazon_web_services.md]: Have updated the description.

**Pull request merged**